### PR TITLE
Users/lesam/announcer2

### PR DESCRIPTION
**Pull request by samclee**
* The "Template-based box" selector can now be used with keyboard controls. Similar to the Rect selector keyboard controls, these controls are enabled by pressing `K` when the Template-based box selector is active.
* Functions related to enabling and disabling keyboard controls are now made public to allow external scripts to incorporate those functions into their own logic.
* The UI elements corresponding to the Template-based box selector have also been updated to use an `AlternatingCrossElement` for higher visibility.
* The announcer that reads the TagsDescriptor after a region is added now reads from the TagsDescriptors `toString()` method instead of only its primary tag.

ex_keyboard_template

### CODE REVIEW

File: src/canvastools/ts/CanvasTools/Region/RegionsManager.ts [hunk 1]
```diff
@@ -161,8 +161,7 @@ export class RegionsManager {
         }
         this.sortRegionsByArea();
         this.redrawAllRegions();
-        document.getElementById("regionAnnouncer").innerHTML =
-            tagsDescriptor.primary ? tagsDescriptor.primary.name : "";
+        document.getElementById("regionAnnouncer").innerHTML = tagsDescriptor.toString();
```
Comment (samclee) [inline src/canvastools/ts/CanvasTools/Region/RegionsManager.ts:164]: Not related to the template selector, this is just an improvement I saw

File: src/canvastools/ts/CanvasTools/Selection/Component/AlternatingCrossElement.ts [hunk 2]
```diff
@@ -54,7 +54,8 @@ export class AlternatingCrossElement extends CrossElement {
      */
     public resize(width: number, height: number) {
         super.resize(width, height);
-        // resize second cross
+        this.vl2.node.setAttribute("y2", height.toString());
```
Comment (samclee) [inline src/canvastools/ts/CanvasTools/Selection/Component/AlternatingCrossElement.ts:57]: This should have been in the last PR. There's not visible change including it but I want it to be consistent

File: src/canvastools/ts/CanvasTools/Selection/Selectors/RectCopySelector.ts [hunk 3]
```diff
@@ -235,4 +231,96 @@ export class RectCopySelector extends Selector {
             this.moveCopyRect(this.copyRectEl, this.crossA);
         });
     }
+
+    /**
+     * Listener for the key down event.
+     * @param e KeyboardEvent
+     */
+    private onKeyDown(e: KeyboardEvent) {
+        if (e.key === "k" || e.key === "K") {
```
Comment (samclee) [inline src/canvastools/ts/CanvasTools/Selection/Selectors/RectCopySelector.ts:240]: At office hours they said double-tapping keys could be troublesome for users with tremors

File: src/canvastools/ts/CanvasTools/Selection/Selectors/RectSelector.ts [hunk 4]
```diff
@@ -389,7 +389,7 @@ export class RectSelector extends Selector {
         return key === "ArrowUp" || key === "ArrowDown" || key === "ArrowLeft" || key === "ArrowRight";
     }
     /**
-     * Helper function for common logic to start a two point selection.
+     * Helper function for common logic to move the keyboard cursor
```
Comment (samclee) [inline src/canvastools/ts/CanvasTools/Selection/Selectors/RectSelector.ts:392]: Unrelated to template selector, an incorrect comment I saw

File: src/canvastools/ts/CanvasTools/Selection/Selectors/RectCopySelector.ts [hunk 5]
```diff
@@ -83,14 +88,35 @@ export class RectCopySelector extends Selector {
         this.showAll([this.crossA, this.copyRectEl]);
     }
 
+    /**
+     *  Returns the member variable `usingKeyboardCursor`
+     */
+    public getUsingKeyboardCursor() {
```
Comment (soyfestivo) [inline src/canvastools/ts/CanvasTools/Selection/Selectors/RectCopySelector.ts:94]: Thanks for making these public, might come in handy if a consumer of vott-ct wants to create their own custom toolbar 😄

File: src/canvastools/ts/CanvasTools/Selection/Selectors/RectCopySelector.ts (hunk 3 above)
Comment (soyfestivo) [reply src/canvastools/ts/CanvasTools/Selection/Selectors/RectCopySelector.ts:261]: That justification makes sense, however my personal opinion would be to use the same key. This is fine.

Small nit:
```typescript
if(e.key.toLocaleUpperCase() === "K") {
  ...
}
```

**Pull request merged**